### DISCUSSION TITLE
Allow more than one search_shortcut

### DISCUSSION THREAD

**Discussion by mbeaty-github** [espanso/espanso#1590] [Feature requests and Ideas]
I move my laptop from a docked setup with an external keyboard, to using the laptop keyboard directly, and it would help to be able to designate more than one search_shortcut to allow for the same ability to use the search pop-up easily regardless of which setup I am using.

**Comment by smeech** [comment]
Sorry this has hung here so long - I'm just trying to help sort out the various Issues and Discussions.

To help me understand your problem, can you explain to me why a single `search_shortcut:` doesn't work for both?

**Reply by mbeaty-github** [reply]
Just because of different configurations, keyboards and devices. 

More specifically on my external keyboard, when docked, I use the hyper (a single key mapped to Shift + Ctrl + Opt + Cmd) +space key, but if I take my MacBook on the go, I don’t have a dedicated hyper key, and I would like to have another combination. 

This is also true using the same config file across Mac and pc and keeping them in sync.

**Reply by smeech** [reply]
Thank you for taking the time to reply.

If you were to use a different key combination, would it then work on both, albeit less conveniently?
Is this a general MacOS issue (in which case I'll tag it as such)?

**Comment by mbeaty-github** [comment]
@smeech sorry for the delayed reply. I could attempt to use the same combination on both status of the laptop (when it was connected to a larger external keyboard, or the laptop keyboard when not docked); however it would be more convenient to allow for more than one input for the `search_shortcut`.

This isn't specifically a MacOS issue, but more so a customization issue? I guess you could try to think of it as if you has the shared matches.yml file sync'd between devices. As another example, if I were to say use CMD + Space on Mac, I would need to use CTRL + Space on PC. 

I was hoping that by allowing more than one combination for `search_shortcut` that I might be able to utilize the search functionality more fluidly between the configuration I have when I'm at my desk vs. when travelling vs. between Mac and PC devices.

Let me know if that makes sense and if I could elaborate further. I am not very familiar with how Rust handles logic like this, but I suppose I am asking if there could be the functionality of an `or` via a `,` or something similar when a User defines the `search_shortcut` value to handle this.

**Comment by smeech** [comment]
I'm not familiar with Rust either, and I'm still not quite sure I understand why you wouldn't use the same shortcut for both, but I'll mark this as a requested enhancement and see what others think.

**Reply by mbeaty-github** [reply]
I guess to put it in the simplest of terms. In this scenario I use a key on my external keyboard. But when I undock my laptop, on my laptop keyboard that key doesn't exist. So I want to be able to assign the shortcut to a key that does exist on the laptop for that situation when it is also undocked.

**Reply by smeech** [reply]
And why don't you assign a key combination that exists on both, for this function? I guess you may use it a lot, so it's convenient to dedicate a key to it?
